File: client/src/main/java/ir/asparsa/hobbytaste/ui/list/holder/LanguageViewHolder.java
package ir.asparsa.hobbytaste.ui.list.holder;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import ir.asparsa.android.ui.fragment.recycler.BaseRecyclerFragment;
import ir.asparsa.android.ui.list.holder.BaseViewHolder;
import ir.asparsa.hobbytaste.R;
import ir.asparsa.hobbytaste.ui.list.data.LanguageData;
import rx.functions.Action1;

/**
 * Created by hadi on 1/13/2017 AD.
 */
public class LanguageViewHolder extends BaseViewHolder<LanguageData> {

    @BindView(R.id.language_layout)
    ViewGroup mLayout;
    @BindView(R.id.language)
    TextView mLanguageTextView;

    public LanguageViewHolder(
            View itemView,
            Action1<BaseRecyclerFragment.Event> observer,
            Bundle savedInstanceState
    ) {
        super(itemView, observer, savedInstanceState);
        ButterKnife.bind(this, itemView);
    }

    @Override public void onBindView(final LanguageData data) {
        mLanguageTextView.setText(data.getLanguage());
        mLayout.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                mObserver.call(new LanguageClick(data.getLangAbbreviation()));
            }
        });
    }

    public static class LanguageClick implements BaseRecyclerFragment.Event {

        private String langAbbreviation;

        LanguageClick(
                String langAbbreviation
        ) {
            this.langAbbreviation = langAbbreviation;
        }

        public String getLangAbbreviation() {
            return langAbbreviation;
        }
    }

}

File: client/src/main/kotlin/ir/asparsa/hobbytaste/ui/list/holder/LanguageViewHolder.kt
package ir.asparsa.hobbytaste.ui.list.holder

import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.ButterKnife
import com.github.kittinunf.reactiveandroid.view.rx_click
import ir.asparsa.android.ui.list.holder.BaseViewHolder
import ir.asparsa.hobbytaste.R
import ir.asparsa.hobbytaste.ui.list.data.LanguageData
import kotterknife.bindView
import rx.Observable

/**
 * @author hadi
 */
class LanguageViewHolder(
        itemView: View
) : BaseViewHolder<LanguageData>(itemView) {

    val mLayout: ViewGroup by bindView(R.id.language_layout)
    val mLanguageTextView: TextView by bindView(R.id.language)

    init {
        ButterKnife.bind(this, itemView)
    }

    override fun onBindView(data: LanguageData) {
        super.onBindView(data)
        mLanguageTextView.text = data.language
    }


    fun clickStream(): Observable<LanguageData> {
        return mLayout.rx_click().map { data }
    }
}
